src/movies.js

```diff
@@ -1,22 +1,64 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    let movie = moviesArray.map ((movie)=> movie.director)
+    return movie
+};
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
-
+function howManyMovies(moviesArray) {
+    if (moviesArray === 0) {
+            return 0
+        }
+        const moviesSpeilberg = moviesArray.filter(allSpielberg => {
+            return allSpielberg.director === "Steven Spielberg" && allSpielberg.genre.includes("Drama")
+        })
+    return moviesSpeilberg.length    
+};
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    const average = moviesArray.reduce((acc, vall) => {
+        return acc + vall.score
+    }, 0)
+    console.log(average)
+    const averageTotal = average / moviesArray.length
+
+    console.log(typeof + averageTotal.toFixed(2))
+        return +averageTotal.toFixed(2)
+};
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    const dramaMovies = moviesArray.filter( movie => movie.genre.includes('Drama'))
+    return scoresAverage(dramaMovies)
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+    const order = moviesArray.splice(0, moviesArray.length)
+    return order.sort(function (a, b) {
+        if (a.year === b.year) {
+            if (a.title < b.title) {
+                return -1
+            } else if (a.title > b.title) {
+                return 1
+            } else {
+                return 0;
+            }
+
+        } return a.year - b.year
+    })
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+    const movieTitles=moviesArray.map (function (movie){
+        return movie.title;
+    })
+    const alphabeticalOrder = movieTitle.sort().slice(0,20);
```

Comment:
Here you are just missing one `s` in `movieTitle`. You declared the `variable` has `movieTitles` but then when you're doing the `.sort()` you missed a `s` and called `movieTitle` so if you go back to your code and just add the `s` your code will pass all the tests